### DISCUSSION TITLE
Indexing UUIDs (DataType::FixedSizeBinary(16)) in BTree

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [lancedb/lancedb#1973] [Q&A] [answered]
I want to index UUIDs for quick search if needed. For optimization purposes, I would like to use DataType::FixedSizeBinary(16), but it seems that this is not supported.

Error:
```bash
A BTree index cannot be created on the field `id` which has data type FixedSizeBinary(16)
```

Do you think it's a good idea to implement it? Or is there another way to do it?

**Comment by albertlockett** [comment, marked as answer]
Hi @vincent-herlemont! Unfortunately, creating BTree indices on this index data type is not currently supported. I don't think there's any reason why we couldn't support this in the future. I've created an issue for us to support this: https://github.com/lancedb/lancedb/issues/1994

Currently, BTree indices are only supported on integers, floats, these datatypes:
```
DataType::Boolean
DataType::Utf8
DataType::Time32
DataType::Time64
DataType::Date32
DataType::Date64
DataType::Timestamp
```

**Comment by LuQQiu** [comment]
Hi @vincent-herlemont, FixedSizeBinary is not in the supported list of datafusion min/max calculator which we used for creating the BTree. 
https://github.com/apache/datafusion/blob/54a5d3fd3f98997f048c2e3e7ba67d7bfe2f124d/datafusion/functions-aggregate/src/min_max.rs#L518-L520
